primitives/Cargo.toml

```diff
@@ -16,9 +16,18 @@ exclude = ["tests", "contrib"]
 
 [features]
 default = ["std"]
-std = []
+std = ["internals/std", "io/std", "units/std"]
+serde = ["actual-serde", "internals/serde", "units/serde"]
```

Comment:
Potentially add an `alloc` feature absence of which just disables the whole crate, so we can stably expose some parts as no-alloc in the future.

Reply:
Very interesting idea. concept ACK from me. Probably should go in its own PR so that people searching the git history don't find that such a big thing showed up as a side-effect of a tiny PR like this.

We could also do the same for base58 and possibly others.

Reply:
If by "big" you mean number of `cfg` lines then there's a lazy version: `#[cfg(not(feature = "alloc"))] compile_error!("no-alloc is not supported yet");` Might be even clearer to get that message rather than "`Transaction` not found in `primitives`".

Reply:
Please see #2996. Note I used `#![cfg(feature = "alloc")]` otherwise we can't build with `--no-default-features` and the CI script breaks (`run_task.sh`).

Reply:
By "big" I mean number of features that are required for literally every line of code in the lib :P.

And yes, I think adding a `compile_error!` while the feature is totally required would be kind. Hopefully we can remove it soon. I think that all the keytypes, blockheaders, version, sequences and locktimes (if those are not `units`) can be no-alloc.

Reply:
Turns out that we do not need that many alloc feature gates, this PR now includes 12 of them in `primitives` mostly in `sequence`. And `locktime` module is out all together if `alloc` is not enabled.